validation/groups.rst

```diff
@@ -185,3 +185,13 @@ will be applied.
 You'll usually work with validation indirectly through the form
 library. For information on how to use validation groups inside forms, see
 :doc:`/form/validation_groups`.
+
+.. caution::
+
+    When using groups on a composite constraint (like ``All`` and
+    ``Collection``), the groups will propagate to the nested constraints,
+    unless some groups are explicitely defined on these.
+
+    However, note that using a composite constraint inside another one will
```

Comment:
IIRC @xabbuh proposed a PR recently which fixes this behavior, or am I wrong? 🤨

Reply:
To be honest, reading this note I have a hard time spinning my head around this.

@nesk Could you explain with an example?

Reply:
It happened to me with the following code:

```php
class User
{
    /**
     * @var string[]
     *
     * @Assert\All({
     *     @Assert\NotBlank,
     *     @Assert\Sequentially({
     *         @Assert\Type("string"),
     *         @AppAssert\ValidPermission,
     *     }, groups={"permissions"}),
     * }, groups={"permissions"})
     */
    private $permissions = [];
}
```

The `Sequentially` composite constraint doesn't inherit the groups of the `All` composite constraint.

Reply:
What is the result with your example? Is only one of the constraints validated?

Reply:
If I try to validate my entity with the `permissions` group, only the `All` and `NotBlank` constraints are applied.

Reply:
I think we should discuss in the code repository first what the expected behaviour should be here. This may as well be a bug instead.